components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/HistoryMetadataSuggestionProvider.kt

```diff
@@ -15,28 +15,28 @@ import mozilla.components.concept.storage.HistoryStorage
 import mozilla.components.feature.session.SessionUseCases
 import java.util.UUID
 
-internal const val METADATA_SUGGESTION_LIMIT = 5
+internal const val DEFAULT_METADATA_SUGGESTION_LIMIT = 5
 
 /**
  * A [AwesomeBar.SuggestionProvider] implementation that provides suggestions based on [HistoryMetadata].
  *
- * @property historyStorage an instance of the [HistoryStorage] used
+ * @param historyStorage an instance of the [HistoryStorage] used
  * to query matching metadata records.
- * @property loadUrlUseCase the use case invoked to load the url when the
+ * @param loadUrlUseCase the use case invoked to load the url when the
  * user clicks on the suggestion.
- * @property icons optional instance of [BrowserIcons] to load fav icons
+ * @param icons optional instance of [BrowserIcons] to load fav icons
  * for [HistoryMetadata] URLs.
  * @param engine optional [Engine] instance to call [Engine.speculativeConnect] for the
  * highest scored suggestion URL.
- * @param maxNumberOfResults optional parameter allowing to lower the number of returned suggested
- * history items to below the default of 5
+ * @param maxNumberOfSuggestions optional parameter allowing to specify a different number than the
+ * default of 5 for how many history results to be returned.
```

Comment:
nit: optional parameter to specify the maximum number of returned suggestions, defaults to [DEFAULT_METADATA_SUGGESTION_LIMIT].